lib/rucio/daemons/hermes/hermes.py

```diff
@@ -26,109 +27,95 @@
+                                 thread=worker_number,
+                                 total_threads=total_workers,
+                                 event_type='email')
 
-                msg = MIMEText(message['payload']['body'])
-                msg['From'] = email_from
-                msg['To'] = ', '.join(message['payload']['to'])
-                msg['Subject'] = message['payload']['subject']
+    if messages != []:
+        to_delete = []
+        for message in messages:
+            logger(logging.DEBUG, 'submitting: %s', str(message))
 
-                if send_email:
-                    smtp = smtplib.SMTP()
-                    smtp.connect()
-                    smtp.sendmail(msg['From'], message['payload']['to'], msg.as_string())
-                    smtp.quit()
+            msg = MIMEText(message['payload']['body'])
+            msg['From'] = email_from
+            msg['To'] = ', '.join(message['payload']['to'])
+            msg['Subject'] = message['payload']['subject']
 
-                to_delete.append({'id': message['id'],
-                                  'created_at': message['created_at'],
-                                  'updated_at': message['created_at'],
-                                  'payload': str(message['payload']),
-                                  'event_type': 'email'})
-
-                logger(logging.DEBUG, 'submitting done: %s',
-                       str(message['id']))
+            if send_email:
+                smtp = smtplib.SMTP()
+                smtp.connect()
+                smtp.sendmail(msg['From'], message['payload']['to'], msg.as_string())
+                smtp.quit()
 
-            delete_messages(to_delete)
-            logger(logging.INFO, 'submitted %i messages', len(to_delete))
+            to_delete.append({'id': message['id'],
+                              'created_at': message['created_at'],
+                              'updated_at': message['created_at'],
+                              'payload': str(message['payload']),
+                              'event_type': 'email'})
 
-        if once:
-            break
+            logger(logging.DEBUG, 'submitting done: %s',
+                   str(message['id']))
 
-        if len(messages) < bulk:
-            logger(logging.INFO, "Only %d messages, which is less than the bulk %d, will sleep"
-                   % (len(messages), bulk))
-            daemon_sleep(start_time=t_start, sleep_time=sleep_time, graceful_stop=GRACEFUL_STOP)
-
-    logger(logging.DEBUG, 'graceful stop requested')
-
-    die(executable, hostname, pid, heartbeat_thread)
-
-    logger(logging.DEBUG, 'graceful stop done')
+        delete_messages(to_delete)
+        logger(logging.INFO, 'submitted %i messages', len(to_delete))
 
+    if len(messages) < bulk:
+        logger(logging.INFO, "Only %d messages, which is less than the bulk %d, will sleep"
+               % (len(messages), bulk))
+        must_sleep = True
+        return must_sleep
```

Comment:
return False. There is a difference between None and False. On None it will sleep

Reply:
then I don't understand the significance of `must_sleep = True` :-D

Reply:
True = True
False = False
None = [True](https://github.com/rucio/rucio/blob/51ca4ed4bb3d3458bedb3b05e67ae023811a0139/lib/rucio/daemons/common.py#L143)

But I'm both hands for changing all daemons which rely on this implicit behavior to make it explicit and remove this "None" behavior

Reply:
I think you missunderstood me. 
I was speaking about returning "False" when it must not sleep. rather than relying on the implicit "return None" at the end of the function 
```
    must_sleep = False
    if len(messages) < bulk:
        must_sleep = True
    return must_sleep
```